### PR TITLE
Fix pyarrow test

### PR DESCRIPTION
**Pull request by mvanschellebeeck**
This fixes [#4448](https://github.com/apache/arrow-datafusion/issues/4448)

The last successful pyarrow test on master ran [here](https://github.com/apache/arrow-datafusion/actions/runs/3586399230/jobs/6035683574) on **ubuntu-20.04** and the most recent failure ran [here](https://github.com/apache/arrow-datafusion/actions/runs/3586606486/jobs/6036118052) on **ubuntu-22.04**

`ubuntu-latest` is starting a transition from 20.04 to 22.04 and it looks like it just hit - see https://github.blog/changelog/2022-11-09-github-actions-ubuntu-latest-workflows-will-use-ubuntu-22-04/

### CODE REVIEW

**Review by jackwener — APPROVED** [review]
LGTM, Thanks @mvanschellebeeck 
Fixing this version is a good choice, using a rolling upgrade version can easily cause dependency problem.

**Review by xudong963 — APPROVED** [review]

**Comment by alamb** [comment]
Thank you @mvanschellebeeck !

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -296,7 +296,7 @@ jobs:
   test-datafusion-pyarrow:
     name: cargo test pyarrow (amd64)
     needs: [linux-build-lib]
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
```
Comment (findepi) [inline .github/workflows/rust.yml:299]: Pinning the image is only a temporary solution.
Obviously GitHub is not going to host those pinned images for ever.
Is there a follow up ticket to fix this?

Comment (alamb) [reply .github/workflows/rust.yml:299]: Filed https://github.com/apache/datafusion/issues/14776

**Pull request merged**